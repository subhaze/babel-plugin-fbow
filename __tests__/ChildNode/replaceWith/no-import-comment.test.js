const babel = require("babel-core");
const plugin = require("../../../index.js");

const testCode = `
// @no-fbow
var parent = document.createElement("div");
var child = document.createElement("p");
parent.appendChild(child);
var span = document.createElement("span");

child.replaceWith(span);
span.replaceWith(child);
`;

it("adds import for replaceWith once", () => {
	const { code } = babel.transform(testCode, {
		plugins: [plugin],
	});
	expect(code).toMatchSnapshot();
});
