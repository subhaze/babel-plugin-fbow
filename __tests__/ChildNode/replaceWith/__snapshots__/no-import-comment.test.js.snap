// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adds import for replaceWith once 1`] = `
"
// @no-fbow
var parent = document.createElement(\\"div\\");
var child = document.createElement(\\"p\\");
parent.appendChild(child);
var span = document.createElement(\\"span\\");

child.replaceWith(span);
span.replaceWith(child);"
`;
